pymeasure/display/windows.py

```diff
@@ -389,19 +400,24 @@ def open_experiment(self):
                 else:
                     results = Results.load(filename)
                     experiment = self.new_experiment(results)
-                    experiment.curve.update()
+                    for index in range(len(self.widget_list)):
```

Comment:
Why not just iterate over `experiment.curve`?

Reply:
I agree